client.go

```diff
@@ -370,6 +370,7 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 }
 
 func (c *Client) handleError(resp *http.Response) (*http.Response, error) {
+	defer resp.Body.Close()
```

Comment:
@NouemanKHAL we saw this fix for out current issue we've been facing. Thank you for diving into this !!

We were wondering if it might be more error proof to ensure `resp` is not nil before calling methods on it?

Like so:

```go
if resp != nil {
  defer resp.Body.Close()
}
```

Hope this helps!

Reply:
Hi @basiszwo,

Glad to hear that :) 

And yes, that would be safer, however in the current implementation, the `handleError` function returns an error response regardless of the status code. So, it's always expected to be called after checking the `resp.StatusCode` property, and therefore `resp` would not be `nil`.

```go
if resp.StatusCode >= http.StatusBadRequest {
		return c.handleError(resp)
}
```